### ISSUE TITLE
8$: error on parenthesized parameter

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#852]
These lines are invalid, but quick-lint-js doesn't diagnose any of them:
```javascript
let g = ((x)) => { };
let f = function ((x)) { };
```
Report an error if a parameter is parenthesized.

We probably just need to create a new diagnostic type and add a call to `diag_reporter_->report` in `parser::visit_binding_element`.

**Comment by factscub** [comment]
i would like to work on this issue. please assign it to me.

**Comment by samdickson** [comment]
Could I take a look at fixing this issue?

**Comment by strager** [comment]
I think here is where we would report the diagnostic:
https://github.com/quick-lint/quick-lint-js/blob/261bec8151f1c0a322241be66604feee9094120a/src/quick-lint-js/fe/parse-statement.cpp#L4054-L4058

**Comment by strager** [comment]
This place too:
https://github.com/quick-lint/quick-lint-js/blob/261bec8151f1c0a322241be66604feee9094120a/src/quick-lint-js/fe/parse-expression.cpp#L2082-L2085

**Comment by samdickson** [comment]
Thanks

**Comment by strager** [comment]
> ([(x,)] => {}); causing both error E0180 (stray comma in function parameter) and new error E0349. How should I go about it.

I think it's fine to report both E0180 and E0349 for this code. Feel free to modify that test.
